src/topk.c

```diff
@@ -119,22 +120,25 @@ char *TopK_Add(TopK *topk, const char *item, size_t itemlen) {
         countPtr = &runner->count;
         if(*countPtr == 0) {
             runner->fp = fp;
-            *countPtr = 1;
+            *countPtr = increment;
             maxCount = max(maxCount, *countPtr);
         } else if(runner->fp == fp) {
             if(itemHeapPtr || *countPtr <= heapMin) {
-                ++*countPtr;
+                *countPtr += increment;
                 maxCount = max(maxCount, *countPtr);                                                     
             }
         } else {
-            double decay = pow(topk->decay, *countPtr);
-            double chance = rand() / (double)RAND_MAX;
-            if(chance < decay) {
-                --*countPtr;
-                if(*countPtr == 0) {
-                    runner->fp = fp;
-                    *countPtr = 1;
-                    maxCount = max(maxCount, *countPtr);
+            uint32_t local_incr = increment;
+            for(; local_incr > 0; --local_incr) {
```

Comment:
You can declare `local_incr` inline e.g.

```
for (uint32_t local_incr = increment; local_inc > 0; --local_incr)
```

Reply:
@mnunberg 
I know. But since it isn't just an index and sometimes the reminder of it is used to increase the counter as a chunk (line 139), I thought it will be clearer.